### PR TITLE
Clean security groups if api/ssh ips are removed from config

### LINKED ISSUE #7125: IPs in k8s api / ssh is not cleaned if removed from config
**1. What `kops` version are you running? The command `kops version`, will display
 this information.** master

**2. What Kubernetes version are you running? `kubectl version` will print the
 version if a cluster is running or provide the Kubernetes version specified as
 a `kops` flag.**

kops master, kubectl 1.14.1

**3. What cloud provider are you using?**

openstack

**4. What commands did you run?  What is the simplest way to reproduce this issue?**

I had following (just for test) in kops config:

```
  kubernetesApiAccess:
  - 0.0.0.0/0
  - 8.8.8.8/32
  sshAccess:
  - 0.0.0.0/0
  - 8.8.8.8/32
```

I modified it to:
```
  kubernetesApiAccess:
  - 0.0.0.0/0
  sshAccess:
  - 0.0.0.0/0
```
and executed kops update cluster (does not print any changes, and when running with --yes it does not update anything)

**5. What happened after the commands executed?**

Nothing, it did not remove anything

**6. What did you expect to happen?**

I expect that rules that are removed from config, are removed from openstack as well.

/sig openstack

### PR DESCRIPTION
**Pull request by zetaab**
fixes #7125

### CODE REVIEW

File: pkg/model/openstackmodel/firewall.go [hunk 1]
```diff
@@ -479,6 +480,13 @@ func (b *FirewallModelBuilder) Build(c *fi.ModelBuilderContext) error {
 			Name:      s(groupName),
 			Lifecycle: b.Lifecycle,
 		}
+		if role == kops.InstanceGroupRoleBastion {
+			sg.RemoveExtraRules = []string{"port=22"}
+		} else if role == kops.InstanceGroupRoleNode {
+			sg.RemoveExtraRules = []string{"port=22", "port=10250"}
```
Comment (zetaab) [inline pkg/model/openstackmodel/firewall.go]: @drekle do you have idea why 10250 (kubelet port) is opened for all ips that are in kubernetesapiaccess? IMO that is not needed

Comment (zetaab) [reply pkg/model/openstackmodel/firewall.go]: https://github.com/kubernetes/kops/blob/master/pkg/model/openstackmodel/firewall.go#L273-L295

**Comment by zetaab** [comment]
/hold

wait until we have answer to kubelet port question

**Comment by zetaab** [comment]
@justinsb @mikesplain something wrong with gomod check

```
I0912 05:55:52.896] ERROR: go modules are not up to date; please run: go mod tidy
I0912 05:55:52.897] changed files:
I0912 05:55:52.897] ?? .config/
I0912 05:55:52.897] git diff:
W0912 05:55:52.931] Traceback (most recent call last):
W0912 05:55:52.931]   File "/workspace/./test-infra/jenkins/../scenarios/execute.py", line 50, in <module>
W0912 05:55:52.932]     main(ARGS.env, ARGS.cmd + ARGS.args)
W0912 05:55:52.932]   File "/workspace/./test-infra/jenkins/../scenarios/execute.py", line 41, in main
W0912 05:55:52.932]     check(*cmd)
W0912 05:55:52.932]   File "/workspace/./test-infra/jenkins/../scenarios/execute.py", line 30, in check
W0912 05:55:52.932]     subprocess.check_call(cmd)
W0912 05:55:52.932]   File "/usr/lib/python2.7/subprocess.py", line 186, in check_call
W0912 05:55:52.932]     raise CalledProcessError(retcode, cmd)
W0912 05:55:52.932] subprocess.CalledProcessError: Command '('./hack/verify-gomod',)' returned non-zero exit status 1
```

I have not created folder called `.config`

**Comment by mikesplain** [comment]
@zetaab, I'll open a cherry pick to 1.15, should this be in 1.14 as well?

**Comment by zetaab** [comment]
@mikesplain yes I think it could be cherrypicked to 1.14 as well. This is basically bug fix, because now the behaviour is that rules are not cleaned. It may lead to situations that there are sg rules that should not be there (like kops is opening 0.0.0.0 by default if you do not specify anything when creating cluster)

**Pull request merged**